tests/framework/extensions/clusters/aks/resources/aks_cluster.go

```diff
@@ -0,0 +1,106 @@
+package resources
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+)
+
+// CreateAKSCluster creates an AKS cluster in the specified resource group.
+func CreateAKSCluster(aksHostCluster *management.AKSClusterConfigSpec, aksImportCluster *management.AKSImportedClusterConfigSpec, clusterName, subscriptionID, clientID string, credential azcore.TokenCredential) (*runtime.Poller[armcontainerservice.ManagedClustersClientCreateOrUpdateResponse], error) {
```

Comment:
So Markus, this might have bee lost in discussion, but I thought we agreed that the actual AKS cluster creation would happen instead in corral. That way we aren't supporting multiple node provider SDKs in the framework. Unless others disagree,  I think the actual cluster creation should happen in corral instead. Anyone else please weigh in.

Reply:
Correct, corral was discussed as the solution. I had the solution presented here in short-term as we are prepping for k8s 1.25, that will take highest priority on our end for testing.

It will take a bit of ramp-up time on our end with Corral as well, so I had this in the short-term. I can hold off if we believe that is the best case to have that implemented now rather than later.

Reply:
Ah ok, well I don't think a short term solution is the answer. We need to decide on how we want to support this. I think it would be best to have it done in corral in order to avoid supporting multiple node provider SDKs, that was brought up as a concern from the dev team, in the initial designing of the framework. But also I can be persuaded otherwise, if multiple people think it would be to do it here.